Common/WALinuxAgent-2.0.16/waagent

```diff
@@ -6509,10 +6509,13 @@ def DistInfo(fullname=0):
         distinfo[0] = distinfo[0].strip()  # remove trailing whitespace in distro name
         if not distinfo[0]:
             distinfo = dist_info_SLES15()
+        if not distinfo[0]:
+            distinfo = dist_info_opensuse15()
```

Comment:
This fix is needed because the library we use to determine the OS and distro fails for newer distros which stopped having release files like they used to, e.g. /etc/suse-release has been dropped in newer distors in favor of /etc/os-release.

We have had to add such methods do deal with these cases since we can't rely on one single method to determine distinfo